server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodesInfoRequest.java

```diff
@@ -208,61 +199,116 @@ public NodesInfoRequest http(boolean http) {
      * @return The request
      */
     public NodesInfoRequest plugins(boolean plugins) {
-        this.plugins = plugins;
+        addOrRemoveMetric(plugins, Metrics.PLUGINS.metricName());
         return this;
     }
 
     /**
      * @return true if information about plugins is requested
      */
     public boolean plugins() {
-        return plugins;
+        return Metrics.PLUGINS.containedIn(requestedMetrics);
     }
 
     /**
      * Should information about ingest be returned
      * @param ingest true if you want info
      */
     public NodesInfoRequest ingest(boolean ingest) {
-        this.ingest = ingest;
+        addOrRemoveMetric(ingest, Metrics.INGEST.metricName());
         return this;
     }
 
     /**
      * @return true if information about ingest is requested
      */
     public boolean ingest() {
-        return ingest;
+        return Metrics.INGEST.containedIn(requestedMetrics);
     }
 
     /**
      * Should information about indices (currently just indexing buffers) be returned
      * @param indices true if you want info
      */
     public NodesInfoRequest indices(boolean indices) {
-        this.indices = indices;
+        addOrRemoveMetric(indices, Metrics.INDICES.metricName());
         return this;
     }
 
     /**
      * @return true if information about indices (currently just indexing buffers)
      */
     public boolean indices() {
-        return indices;
+        return Metrics.INDICES.containedIn(requestedMetrics);
+    }
+
+    /**
+     * Helper method for adding and removing metrics.
+     * @param includeMetric Whether or not to include a metric.
+     * @param metricName Name of the metric to include or remove.
+     */
+    private void addOrRemoveMetric(boolean includeMetric, String metricName) {
```

Comment:
I'm in two minds over whether this is a better name. WDYT?

```suggestion
    private void toggleMetric(boolean includeMetric, String metricName) {
```

Reply:
toggle doesn't make sense, since the new state is passed in, not flipped.

Reply:
I'll see if I can make this awkwardly named method go away when I change this class's public API in an upcoming PR.